### ISSUE TITLE
Saving data at the end of dataset (i.e. timeseries data)

### ISSUE THREAD

**Issue by wrobell** [zarr-developers/zarr-python#473]
I am considering to use Zarr for saving timeseries-like data.

By `timeseries-like` I mean that an application receives data and appends it to the end of a dataset. This can be used to store timeseries data, but it might cover other use cases.

My understanding is, that if I save data to the end of a dataset, then Zarr will
- read and uncompress last chunk (if I am lucky it will get it from cache?)
- then compress and save last chunk

I wonder if it could be possible to instruct Zarr to
- never compress *last* chunk
- save data at the end of the dataset (only new data, without rewriting whole chunk)

**Comment by joshmoore** [comment]
Do you have an example of the chunk and array sizes you'd like to use? e.g. if saves are always performed on chunk boundaries, then no re-reading would be necessary.

**Comment by wrobell** [comment]
Let's say you want to store GPS data. Your chunk could be around 3600 (one hour) or 24 * 3600 (a day) worth of data. But you risk loosing the data when keeping it in memory until a chunk fills.

I did a bit of thinking and research in last couple of weeks, the issue is not trivial. I will spill my thoughts into this project https://gitlab.com/wrobell/n23ts (see also https://gitlab.com/wrobell/n23ts/blob/master/doc/zarr.rst). Closing for now.
